server/mqtt.go

```diff
@@ -1667,6 +1729,32 @@ func (jsa *mqttJSA) loadLastMsgFor(streamName string, subject string) (*StoredMs
 	return lmr.Message, lmr.ToError()
 }
 
+func (jsa *mqttJSA) loadLastMsgForMulti(streamName string, subjects []string) ([]*JSApiMsgGetResponse, error) {
+	marshaled := [][]byte{}
+	headerBytes := []int{}
+	for _, subject := range subjects {
+		mreq := &JSApiMsgGetRequest{LastFor: subject}
+		bb, err := json.Marshal(mreq)
+		if err != nil {
+			return nil, err
+		}
+		marshaled = append(marshaled, bb)
+		headerBytes = append(headerBytes, 0)
+	}
+
+	all, err := jsa.newRequestExMulti(mqttJSAMsgLoad, fmt.Sprintf(JSApiMsgGetT, streamName), _EMPTY_, headerBytes, marshaled, mqttJSAPITimeout)
+	if err != nil {
```

Comment:
So here, I would not check error since `all` would always be returned with possibly some responses present, some nil.
```
responses := make([]*JSApiMsgGetResponse, len(all))
for i, v := range all {
	if v != nil {
		responses[i] = v.value.(*JSApiMsgGetResponse)
	}
}
return responses, err
```

Reply:
And `loadRetainedMessages` ignores the error now, loading and then delivering whatever messages came through.